nmos-test.py

```diff
@@ -422,54 +423,78 @@ def init_spec_cache():
     print(" * Initialisation complete")
 
 
-def write_test_results(results, args):
-    exit_code = ExitCodes.OK
-    test_cases = []
-    for test_result in results["result"]:
-        test_case = TestCase(test_result.name, classname=results["def"]["class"].__name__,
-                             elapsed_sec=test_result.elapsed_time, timestamp=test_result.timestamp)
-        if test_result.name in args.ignore or test_result.state in [TestStates.DISABLED,
-                                                                    TestStates.UNCLEAR,
-                                                                    TestStates.MANUAL,
-                                                                    TestStates.NA,
-                                                                    TestStates.OPTIONAL]:
-            test_case.add_skipped_info(test_result.detail)
-        elif test_result.state in [TestStates.WARNING, TestStates.FAIL]:
-            test_case.add_failure_info(test_result.detail, failure_type=str(test_result.state))
-            if test_result.state == TestStates.FAIL:
-                exit_code = max(exit_code, ExitCodes.FAIL)
-            elif test_result.state == TestStates.WARNING:
-                exit_code = max(exit_code, ExitCodes.WARNING)
-        elif test_result.state != TestStates.PASS:
-            test_case.add_error_info(test_result.detail, error_type=str(test_result.state))
-        test_cases.append(test_case)
-
-    ts = TestSuite(results["def"]["name"] + ": " + results["base_url"], test_cases)
-    with open(args.output, "w") as f:
-        # pretty-print to help out Jenkins (and us humans), which struggles otherwise
-        TestSuite.to_file(f, [ts], prettyprint=True)
-        print(" * Test results written to file: {}".format(args.output))
-    return exit_code
-
-
-def print_test_results(results, args):
+def format_test_results(results, format):
+    formatted = None
+    if format == "json":
+        formatted = {"suite": results["def"]["name"],

```

Comment:
Would the test suite id (as required for the `suite` sub-command, or returned from `--list-suites`) be useful instead/as well?

(Though a similar remark could be applied to the JUnit output which uses `results["def"]["class"].__name__`.)

Reply:
Agreed, I've modified this in all cases.